src/xdp/xsk.c

```diff
@@ -3945,7 +3968,8 @@ XskNotifyValidateParams(
     }
 
     if (*InFlags == 0 || *InFlags &
-            ~(XSK_NOTIFY_FLAG_POKE_RX | XSK_NOTIFY_FLAG_POKE_TX | XSK_NOTIFY_FLAG_WAIT_RX | XSK_NOTIFY_FLAG_WAIT_TX)) {
+            ~(XSK_NOTIFY_FLAG_POKE_RX | XSK_NOTIFY_FLAG_POKE_TX | XSK_NOTIFY_FLAG_WAIT_RX | XSK_NOTIFY_FLAG_WAIT_TX |
+              XSK_NOTIFY_FLAG_WAKE)) {
         return STATUS_INVALID_PARAMETER;
```

Comment:
It should be invalid to set XSK_NOTIFY_FLAG_WAKE along with any XSK_NOTIFY_FLAG_WAIT_* flag, right?

Reply:
Or can the cancel be chained to a wait? If so, the latter needs to be tested.

Thread A: waits.
Thread B: cancels A's wait and starts waiting itself.

Is this a useful scenario?

Reply:
Maybe not? Maybe change to the current thread being the wait thread? IDK...

Reply:
Yeah, it's kind of weird and probably useless, but I don't see a reason to specifically exclude this combination of flags from a technical point of view. The behavior is well-defined, just not terribly useful.